### PR TITLE
Microkit

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1197.

### CODE REVIEW

File: server/endpoint/error.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"github.com/juju/errgo"
+)
+
+var invalidConfigError = errgo.New("invalid config")
+
+// IsInvalidConfig asserts invalidConfigError.
+func IsInvalidConfig(err error) bool {
+	return errgo.Cause(err) == invalidConfigError
+}
```
Comment (kopiczko) [inline server/endpoint/error.go]: I think we could move this error to microkit at some point.

File: service/service.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	}
+}
+
+// New creates a new configured service object.
+func New(config Config) (*Service, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "logger must not be empty")
+	}
+	config.Logger.Log("debug", fmt.Sprintf("creating kvm-operator with config: %#v", config))
```
Comment (kopiczko) [inline service/service.go]: kvm-operator

File: server/endpoint/error.go (hunk 1 above)
Comment (xh3b4sd) [inline server/endpoint/error.go]: Uncertain. I would like to move the version endpoint and service to microkit. This is a lot of boilerplate we can safe.

File: service/service.go (hunk 2 above)
Comment (xh3b4sd) [inline service/service.go]: Done.

**Pull request merged**